identity-iota-core/src/diff/diff_iota_document_metadata.rs

```diff
@@ -258,8 +285,43 @@ mod test {
     let original: IotaDocumentMetadata = IotaDocumentMetadata::new();
     let mut updated: IotaDocumentMetadata = IotaDocumentMetadata::new();
     updated.previous_message_id = MessageId::new([1; 32]);
-    updated.created = Timestamp::from_unix(1).unwrap();
-    updated.updated = Timestamp::from_unix(100000).unwrap();
+    updated.created = Some(Timestamp::from_unix(1).unwrap());
+    updated.updated = Some(Timestamp::from_unix(100000).unwrap());
+    let diff: DiffIotaDocumentMetadata = Diff::diff(&original, &updated).unwrap();
+
+    let ser: String = diff.to_json().unwrap();
+    let de: DiffIotaDocumentMetadata = DiffIotaDocumentMetadata::from_json(&ser).unwrap();
+    assert_eq!(diff, de);
+    let merge: IotaDocumentMetadata = Diff::merge(&original, de).unwrap();
+    assert_eq!(merge, updated);
+  }
+
+  #[test]
+  fn test_serde_timestamp_to_none() {
+    let original: IotaDocumentMetadata = IotaDocumentMetadata::new();
+    let mut updated: IotaDocumentMetadata = IotaDocumentMetadata::new();
+    updated.previous_message_id = MessageId::new([1; 32]);
+    updated.created = Some(Timestamp::from_unix(1).unwrap());
+    updated.updated = None;
+    let diff: DiffIotaDocumentMetadata = Diff::diff(&original, &updated).unwrap();
+
+    let ser: String = diff.to_json().unwrap();
+    let de: DiffIotaDocumentMetadata = DiffIotaDocumentMetadata::from_json(&ser).unwrap();
+    assert_eq!(diff, de);
+    let merge: IotaDocumentMetadata = Diff::merge(&original, de).unwrap();
+    assert_eq!(merge, updated);
+  }
+
+  #[test]
+  fn test_serde_timestamp_from_none() {
+    let mut original: IotaDocumentMetadata = IotaDocumentMetadata::new();
+    original.updated = None;
+    original.created = None;
+
+    let mut updated: IotaDocumentMetadata = IotaDocumentMetadata::new();
+    updated.previous_message_id = MessageId::new([1; 32]);
+    updated.created = Some(Timestamp::from_unix(1).unwrap());
+    updated.updated = None;
```

Comment:
Nit: setting the `MessageId` is unnecessary for the timestamp tests.

```suggestion
  #[test]
  fn test_serde_timestamp_to_none() {
    let original: IotaDocumentMetadata = IotaDocumentMetadata::new();
    let mut updated: IotaDocumentMetadata = IotaDocumentMetadata::new();
    updated.created = Some(Timestamp::from_unix(1).unwrap());
    updated.updated = None;
    let diff: DiffIotaDocumentMetadata = Diff::diff(&original, &updated).unwrap();

    let ser: String = diff.to_json().unwrap();
    let de: DiffIotaDocumentMetadata = DiffIotaDocumentMetadata::from_json(&ser).unwrap();
    assert_eq!(diff, de);
    let merge: IotaDocumentMetadata = Diff::merge(&original, de).unwrap();
    assert_eq!(merge, updated);
  }

  #[test]
  fn test_serde_timestamp_from_none() {
    let mut original: IotaDocumentMetadata = IotaDocumentMetadata::new();
    original.updated = None;
    original.created = None;

    let mut updated: IotaDocumentMetadata = IotaDocumentMetadata::new();
    updated.created = Some(Timestamp::from_unix(1).unwrap());
    updated.updated = None;
```